x-pack/plugin/security/qa/multi-cluster/build.gradle

```diff
@@ -6,76 +6,88 @@
-  dependsOn configurations.signedCerts
-  from(configurations.signedCerts)
-  from(configurations.rootCert)
-  into "${buildDir}/certs"
-}
-
-
 def fulfillingCluster = testClusters.register('fulfilling-cluster') {
   requiresFeature 'es.untrusted_remote_cluster_feature_flag_registered', Version.fromString("8.5.0")
+  setting 'xpack.security.enabled', 'true'
   setting 'xpack.license.self_generated.type', 'trial'
   setting 'remote_cluster.enabled', 'true'
   setting 'remote_cluster.port', '0'
 
-  extraConfigFile 'transport.key', file("${buildDir}/certs/n1.c1.key")
-  extraConfigFile 'transport.cert', file("${buildDir}/certs/n1.c1.crt")
-  extraConfigFile 'transport.ca', file("${buildDir}/certs/ca.crt")
+  extraConfigFile 'transport.key', file("src/test/resources/ssl/transport.key")
+  extraConfigFile 'transport.cert', file("src/test/resources/ssl/transport.crt")
+  extraConfigFile 'transport.ca', file("src/test/resources/ssl/transport-ca.crt")
 
-  setting 'xpack.security.transport.ssl.enabled', 'true'
+  // Transport SSL can be enabled or disabled. It is independent from the remote cluster server SSL
+  if (BuildParams.random.nextBoolean()) {
+    setting 'xpack.security.transport.ssl.enabled', 'true'
+  } else {
+    setting 'xpack.security.transport.ssl.enabled', 'false'
+  }
   setting 'xpack.security.transport.ssl.key', 'transport.key'
   setting 'xpack.security.transport.ssl.certificate', 'transport.cert'
+  setting 'xpack.security.transport.ssl.key_passphrase', 'transport-password'
   setting 'xpack.security.transport.ssl.certificate_authorities', 'transport.ca'
   setting 'xpack.security.transport.ssl.client_authentication', 'required'
   setting 'xpack.security.transport.ssl.verification_mode', 'certificate'
 
-  // It is intentionally to use none for both verification_mode and client_authentication.
-  // Because SSL is not wired up properly on the client (QC) side. These settings
-  // just test that they can be configured.
-  // Once SSL is all wired up, we will need (1) proper SSL verification and
-  // (2) different set of key and cert than the ones used for the transport interface.
-  setting 'xpack.security.remote_cluster.ssl.key', 'transport.key'
-  setting 'xpack.security.remote_cluster.ssl.certificate', 'transport.cert'
-  setting 'xpack.security.remote_cluster.ssl.verification_mode', 'none'
-  setting 'xpack.security.remote_cluster.ssl.client_authentication', 'none'
+  // Server side SSL configuration for remote cluster
+  extraConfigFile 'remote-cluster.key', file("src/test/resources/ssl/remote_cluster.key")
+  extraConfigFile 'remote-cluster.cert', file("src/test/resources/ssl/remote_cluster.crt")
+  extraConfigFile 'remote-cluster.ca', file("src/test/resources/ssl/remote-cluster-ca.crt")
+
+  setting 'xpack.security.remote_cluster.ssl.enabled', 'true'
+  setting 'xpack.security.remote_cluster.ssl.key', 'remote-cluster.key'
+  setting 'xpack.security.remote_cluster.ssl.certificate', 'remote-cluster.cert'
```

Comment:
is the certificated needed in the fulfilling cluster ?

Reply:
Yes the server needs a signed certficiate which cannot be derived from just the private key. Our code [prevents node from starting up](https://github.com/elastic/elasticsearch/blob/b4484cd8c62b7080c8038d292975e03fbc03a202/libs/ssl-config/src/main/java/org/elasticsearch/common/ssl/SslConfigurationLoader.java#LL352C4-L352C4) if `key` is configured without a `certificate`.

NOTE: the linked code actually has a bug in the error message: It should report key_path instead of keystore_path. I'll have a separate PR to fix it.